## Tarea grupal — IA en Producción — Grupo 5 ##
### Puntos 1 y 2 — EDA completa y Tablas de resumen ###

**Integrante:** Allison Mora

**Rama:** Allison-Mora



### Carga de librerías y datos

Cargamos librerías necesarias 

In [66]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Establecemos las columnas que se muestran 
pd.set_option('display.max_columns', 30)
# Establecemos un margen para las figuras 
plt.rcParams['figure.figsize'] = (10, 6)

In [67]:
# Nombramos una variable como el .xlsx que utilizaremos 
ARCHIVO = 'plan_de_compras_2025.xlsx'
# Llamamos al .xlsx y llamamos este dataframe como df 
df = pd.read_excel(ARCHIVO, sheet_name='PLAN DE COMPRAS 2025')
#Mostramos las dimensiones de nuestro dataframe
print('Dimensiones del dataset:', df.shape)
# Mostramos las primeras 5 filas de nuestro dataframe
df.head()

Dimensiones del dataset: (831, 26)


,Unidad de Compra,ID Proyecto,Tipo Proyecto,Estado Proyecto,Código presupuestario,Nombre Proyecto,Descripción Proyecto,Cantidad de Ítems,Nombre ítem,Tipo compra,Cantidad Productos,Monto Unitario Ítem,Monto Total Ítem Año 2025,Cantidad OC,Meses envío OC,Nombre responsable,Cargo responsable,Teléfono responsable,Correo responsable,Fecha de Inicio Compra,Fecha Publicación PAC 2025,Cantidad OC Asociadas Ítem 2025,OC Asociada Item 2025,Item De Arraste,Monto De Arrastre,Meses envio OC de Arraste
0,Ministerio de Vivienda y Urbanismo(766),587-259-PC22,Proyecto estratégico,Publicado,2207-2209-33,Servicio arriendo de una plataforma de gestión...,Servicio de arriendo de una plataforma de gest...,2,Servicio arriendo de una plataforma de gestión...,1,1,125000000,20000000,3,"Abr 2023,Ene 2024,Ene 2025",María Teresa Zúñiga Silva,Coordinadora de Tecnologías Informáticas | Sis...,2 -2901 1198-,mzunigas@minvu.cl,2022-11-01,2022-11-08,0,NaN,1,20000000,2025-01-01
1,SEREMI MINVU V REGION,632-28-PC22,Proyecto operacional,Publicado,2206-2206001,Servicio de Mantención y Reparaciones Menores ...,Convenio de Suministro para el Servicio de Man...,1,Mantenimiento y Reparación de Edificaciones,1,1,20000000,4500000,5,"May 2022,Nov 2022,Ene 2023,Ene 2024,Ene 2025",Ariel Gardaix Gardaix,Encargado Sección Administración y Finanzas,32-2186802-,agardaix@minvu.cl,2022-02-01,2022-11-08,1,632-7-SE22,1,4500000,2025-01-01
2,SEREMI MINVU V REGION,632-32-PC22,Proyecto operacional,Publicado,2204001,Servicios de Impresión-Seremi Valparaiso,Servicios de Impresión de Materiales SEREMI y ...,1,Materiales de Oficina,1,1,15300000,3825000,4,"Oct 2022,Ene 2023,Ene 2024,Ene 2025",Ariel Gardaix Gardaix,Encargado Sección Administración y Finanzas,32-2186802-,agardaix@minvu.cl,2022-08-01,2022-11-08,0,NaN,1,3825000,2025-01-01
3,SEREMI MINVU V REGION,632-34-PC22,Proyecto operacional,Publicado,2209006,Servicio de Arriendo de Impresoras Multifuncio...,Servicio de Arriendo de Impresoras Multifuncio...,2,Arriendo de Equipos Informáticos,1,1,18000000,4500000,6,"Oct 2022,Nov 2022,Dic 2022,Ene 2023,Ene 2024,E...",Ariel Gardaix Gardaix,Encargado Sección Administración y Finanzas,32-2186802-,agardaix@minvu.cl,2022-08-01,2022-11-08,1,632-13-CC23,1,4500000,2025-01-01
4,SEREMI MINVU V REGION,632-41-PC22,Proyecto operacional,Publicado,2205005,Servicio de Telefonía Fija-Seremi Valparaiso,Servicio de Telefonía Fija SEREMI,2,Telefonía Fija,1,1,105600000,26400000,4,"Ene 2023,Ene 2024,Ene 2025,Ene 2026",Ariel Gardaix Gardaix,Encargado Sección Administración y Finanzas,322186802,agardaix@minvu.cl,2022-12-01,2022-11-08,0,NaN,1,26400000,2025-01-01


In [ ]:
# Llamamos todos los nombres de nuestras columnas
df.columns.tolist()

['Unidad de Compra',
 'ID Proyecto',
 'Tipo Proyecto',
 'Estado Proyecto',
 'Código presupuestario',
 'Nombre Proyecto',
 'Descripción Proyecto',
 'Cantidad de Ítems',
 'Nombre ítem',
 'Tipo compra',
 'Cantidad Productos',
 'Monto Unitario Ítem',
 'Monto Total Ítem Año 2025',
 'Cantidad OC',
 'Meses envío OC',
 'Nombre responsable',
 'Cargo responsable',
 'Teléfono responsable',
 'Correo responsable',
 'Fecha de Inicio Compra',
 'Fecha Publicación PAC 2025',
 'Cantidad OC Asociadas Ítem 2025',
 'OC Asociada Item 2025',
 'Item De Arraste',
 'Monto De Arrastre',
 'Meses envio OC de Arraste']

In [68]:
#Mostramos el tipo de dato contenido en cada columna de nuestro dataframe
df.dtypes

Unidad de Compra                              str
ID Proyecto                                   str
Tipo Proyecto                                 str
Estado Proyecto                               str
Código presupuestario                      object
Nombre Proyecto                               str
Descripción Proyecto                          str
Cantidad de Ítems                           int64
Nombre ítem                                   str
Tipo compra                                 int64
Cantidad Productos                          int64
Monto Unitario Ítem                         int64
Monto Total Ítem Año 2025                   int64
Cantidad OC                                 int64
Meses envío OC                             object
Nombre responsable                            str
Cargo responsable                             str
Teléfono responsable                       object
Correo responsable                            str
Fecha de Inicio Compra             datetime64[us]


## 1. EDA completa

Exploración inicial: estructura, tipos de datos, estadísticas descriptivas, valores nulos y duplicados.

In [ ]:
#Mostramos un resumen de nuestro dataframe 
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 831 entries, 0 to 830
Data columns (total 26 columns):
 #   Column                           Non-Null Count  Dtype         
---  ------                           --------------  -----         
 0   Unidad de Compra                 831 non-null    str           
 1   ID Proyecto                      831 non-null    str           
 2   Tipo Proyecto                    831 non-null    str           
 3   Estado Proyecto                  831 non-null    str           
 4   Código presupuestario            830 non-null    object        
 5   Nombre Proyecto                  831 non-null    str           
 6   Descripción Proyecto             831 non-null    str           
 7   Cantidad de Ítems                831 non-null    int64         
 8   Nombre ítem                      831 non-null    str           
 9   Tipo compra                      831 non-null    int64         
 10  Cantidad Productos               831 non-null    int64         
 11  Mont

In [ ]:
#funcion calcula estadísticas descriptivas para todas las columnas numéricas y luego transpone el resultado
df.describe(include='number').T

,count,mean,std,min,25%,50%,75%,max
Cantidad de Ítems,831.0,1.199759e+00,4.148449e-01,1.0,1.0,1.0,1.0,3.000000e+00
Tipo compra,831.0,4.657040e+00,1.942318e+00,1.0,4.0,6.0,6.0,6.000000e+00
Cantidad Productos,831.0,1.000000e+00,0.000000e+00,1.0,1.0,1.0,1.0,1.000000e+00
Monto Unitario Ítem,831.0,1.085214e+08,2.036220e+08,90000.0,4200000.0,40000000.0,202000000.0,4.839000e+09
Monto Total Ítem Año 2025,831.0,1.014853e+08,2.024663e+08,0.0,3000000.0,35000000.0,202000000.0,4.839000e+09
Cantidad OC,831.0,1.344164e+00,1.263753e+00,1.0,1.0,1.0,1.0,1.200000e+01
Cantidad OC Asociadas Ítem 2025,831.0,1.257569e+02,1.550410e+02,0.0,1.0,4.0,321.0,3.210000e+02
Item De Arraste,831.0,2.202166e-01,4.146422e-01,0.0,0.0,0.0,0.0,1.000000e+00
Monto De Arrastre,831.0,6.139761e+06,3.466941e+07,0.0,0.0,0.0,0.0,6.747510e+08


In [ ]:
#funcion calcula estadísticas descriptivas para todas las columnas str,txt,etc y luego transpone el resultado
df.describe(include='object').T

/tmp/ipykernel_495/1760094569.py:1: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  df.describe(include='object').T


,count,unique,top,freq
Unidad de Compra,831,16,Ministerio de Vivienda y Urbanismo(766),684
ID Proyecto,831,326,1583-200-PC25,321
Tipo Proyecto,831,2,Proyecto operacional,613
Estado Proyecto,831,2,Actualizado,648
Código presupuestario,830,73,2208007,332
Nombre Proyecto,831,328,ADQUISICIÓN PASAJES - SECCION GESTION DE COMPRAS,321
Descripción Proyecto,831,297,Cubrir la necesidad de traslado de los funcion...,321
Nombre ítem,831,317,ADQUISICIÓN PASAJES - SECCION GESTION DE COMPRAS,321
Meses envío OC,831,59,2025-01-01 00:00:00,339
Nombre responsable,831,85,"MALHUE GUIÑEZ, MARIA ANGELICA",325


In [8]:
# Valores nulos por columna
nulos = pd.DataFrame({
    'nulos': df.isna().sum(),
    'porcentaje_nulos': (df.isna().mean() * 100).round(2)
}).sort_values('nulos', ascending=False)
nulos[nulos['nulos'] > 0]

,nulos,porcentaje_nulos
Meses envio OC de Arraste,648,77.98
OC Asociada Item 2025,123,14.80
Código presupuestario,1,0.12


In [9]:
# Registros duplicados
print('Filas duplicadas:', df.duplicated().sum())

# Cardinalidad de columnas categóricas clave
for col in ['Unidad de Compra', 'Tipo Proyecto', 'Estado Proyecto', 'Tipo compra']:
    print(f"{col}: {df[col].nunique()} valores únicos -> {sorted(df[col].dropna().unique().tolist())}")

Filas duplicadas: 0
Unidad de Compra: 16 valores únicos -> ['Ministerio de Vivienda y Urbanismo(766)', 'SEREMI IV REGION', 'SEREMI IX REGION', 'SEREMI MINVU I REGION', 'SEREMI MINVU II REGION', 'SEREMI MINVU III REGION', 'SEREMI MINVU REGION ARICA Y PARINACOTA', 'SEREMI MINVU RM', 'SEREMI MINVU V REGION', 'SEREMI MINVU VI REGION', 'SEREMI MINVU VII REGION', 'SEREMI MINVU XI REGION', 'SEREMI VIII REGION', 'SEREMI ÑUBLE-REGIÓN XVI', 'SERVICIO DE BIENESTAR MINVU SERVIU', 'Seremi-Minvu XII Región']
Tipo Proyecto: 2 valores únicos -> ['Proyecto estratégico', 'Proyecto operacional']
Estado Proyecto: 2 valores únicos -> ['Actualizado', 'Publicado']
Tipo compra: 6 valores únicos -> [1, 2, 3, 4, 5, 6]


In [10]:
# Rango de montos
df[['Monto Unitario Ítem', 'Monto Total Ítem Año 2025']].agg(['min', 'max', 'mean', 'median'])

,Monto Unitario Ítem,Monto Total Ítem Año 2025
min,9.000000e+04,0.000000e+00
max,4.839000e+09,4.839000e+09
mean,1.085214e+08,1.014853e+08
median,4.000000e+07,3.500000e+07


## 2. Tablas de resumen (4)

Generamos 4 tablas de agregación sobre las dimensiones más relevantes del plan de compras.

### Tabla 1 — Monto total e ítems por Unidad de Compra

In [11]:
tabla1_unidad_compra = (
    df.groupby('Unidad de Compra', dropna=False)
    .agg(n_items=('Nombre ítem', 'count'), monto_total=('Monto Total Ítem Año 2025', 'sum'))
    .sort_values('monto_total', ascending=False)
)
tabla1_unidad_compra

,n_items,monto_total
Unidad de Compra,,
Ministerio de Vivienda y Urbanismo(766),684,81943911910
SEREMI MINVU V REGION,30,1187465518
SEREMI MINVU III REGION,25,309240000
SERVICIO DE BIENESTAR MINVU SERVIU,4,248000000
SEREMI MINVU VI REGION,20,162228246
SEREMI MINVU REGION ARICA Y PARINACOTA,9,135080000
SEREMI MINVU II REGION,8,112600006
SEREMI VIII REGION,6,109333336
SEREMI MINVU RM,3,37427780


### Tabla 2 — Proyectos y montos por Tipo y Estado de Proyecto

In [12]:
tabla2_tipo_estado = (
    df.groupby(['Tipo Proyecto', 'Estado Proyecto'], dropna=False)
    .agg(n_proyectos=('ID Proyecto', 'nunique'), monto_total=('Monto Total Ítem Año 2025', 'sum'))
)
tabla2_tipo_estado

n_proyectos  monto_total
Tipo Proyecto        Estado Proyecto                          
Proyecto estratégico Actualizado               54   2261706290
                     Publicado                 63   4933924934
Proyecto operacional Actualizado              194  76970434876
                     Publicado                 15    168216250

### Tabla 3 — Top 10 responsables por monto total gestionado

In [13]:
tabla3_top_responsables = (
    df.groupby('Nombre responsable', dropna=False)
    .agg(n_items=('Nombre ítem', 'count'), monto_total=('Monto Total Ítem Año 2025', 'sum'))
    .sort_values('monto_total', ascending=False)
    .head(10)
)
tabla3_top_responsables

,n_items,monto_total
Nombre responsable,,
"MALHUE GUIÑEZ, MARIA ANGELICA",325,64851217000
CAVIERES CORDOBA LEONARDO RODRIGO,43,7193980000
"LABBE SALDIAS, CRISTIAN ALFONSO",2,1800000000
"SILVA AGUILAR, VICTOR MANUEL",7,1653517298
"San Martín Henríquez, Alejandra Lorena",5,1110000000
"Gascon Huerta, Luis Humberto",36,972894147
CARLOS ANDRES ROJAS ASTORGA,14,966355666
"Rudloff Kuttner, Cristian Heriberto",2,800000000
Elpida Ibañez Tapia,22,469866668


### Tabla 4 — Resumen por Tipo de compra

In [14]:
tabla4_tipo_compra = (
    df.groupby('Tipo compra', dropna=False)
    .agg(
        n_items=('Nombre ítem', 'count'),
        monto_promedio=('Monto Unitario Ítem', 'mean'),
        monto_total=('Monto Total Ítem Año 2025', 'sum'),
    )
    .sort_values('monto_total', ascending=False)
)
tabla4_tipo_compra

,n_items,monto_promedio,monto_total
Tipo compra,,,
6,486,1.399832e+08,67870307246
5,79,1.038030e+08,7440038312
1,159,5.460496e+07,4921854535
4,83,5.970584e+07,3921795608
2,4,4.720970e+07,123733890
3,20,6.119523e+06,56552759


## Resumen de lo obtenido 

- **EDA:** 831 registros, 26 columnas originales, sin duplicados exactos; nulos concentrados en teléfono/correo/campos de arrastre.
- **Tablas de resumen:** por unidad de compra, tipo/estado de proyecto, top responsables y tipo de compra.
